### PR TITLE
reduce use of preconditions and asserts

### PR DESCRIPTION
**Pull request by tomerd**
motivation: better robustness

changes:replace use of preconditions and asserts with throwing guards

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Review by abertelrud — APPROVED** [review]

File: Sources/Build/BuildPlan.swift [hunk 1]
```diff
@@ -870,7 +875,9 @@ public final class SwiftTargetBuildDescription {
 
     /// Command-line for emitting just the Swift module.
     public func emitModuleCommandLine() throws -> [String] {
-        assert(buildParameters.emitSwiftModuleSeparately)
+        guard buildParameters.emitSwiftModuleSeparately else {
+            throw InternalError("expecting emitSwiftModuleSeparately build")
```
Comment (neonichu) [inline Sources/Build/BuildPlan.swift:879]:
```suggestion
            throw InternalError("expecting emitSwiftModuleSeparately in build parameters")
```

File: Sources/Build/BuildPlan.swift [hunk 2]
```diff
@@ -915,7 +922,9 @@ public final class SwiftTargetBuildDescription {
     ///
     /// Note: This doesn't emit the module.
     public func emitObjectsCommandLine() throws -> [String] {
-        assert(buildParameters.emitSwiftModuleSeparately)
+        guard buildParameters.emitSwiftModuleSeparately else {
+            throw InternalError("expecting emitSwiftModuleSeparately build")
```
Comment (neonichu) [inline Sources/Build/BuildPlan.swift:926]:
```suggestion
            throw InternalError("expecting emitSwiftModuleSeparately in build parameters")
```

**Review by neonichu — APPROVED** [review]

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by tomerd** [comment]
@swift-ci please smoke test linux

**Pull request merged**